dashboard/app/models/user.rb

```diff
@@ -835,8 +835,8 @@ def User.track_proficiency(user_id, script_id, level_id)
 
   # returns whether a new level has been completed and asynchronously enqueues an operation
   # to update the level progress.
-  def track_level_progress_async(script_level:, new_result:, submitted:, level_source_id:)
-    level_id = script_level.level_id
+  def track_level_progress_async(script_level:, new_result:, submitted:, level_source_id:, level:)
+    level_id = level.try(:id) || script_level.level_id
```

Comment:
Yeah, I actually had this same change on a local branch.

For my case, the progress has to be saved on the underlying level, since there's no parent level. I need to change the rendering a bit to basically take a `max` of the completion status of the underlying levels.

Are you planning to save completion status for both the underlying and parent levels? If not, the rendering logic would get more complicated.